src/OneExplorer/OneExplorer.ts

```diff
@@ -176,7 +176,13 @@ abstract class Node {
 }
 
 class NodeFactory {
-  static create(type: NodeType, fpath: string, attr?: ArtifactAttr): Node {
+  static create(type: NodeType, fpath: string, attr?: ArtifactAttr): Node|undefined {
+    // TreeDataProvider expects every elements (Node) to be correspond to visible TreeItem, so let's not build hidden nodes.
+    if(attr?.canHide === true && OneTreeDataProvider.didHideExtra === true)
```

Comment:
For reviewers,

To refer here, `OneTreeDataProvider.didHideExtra` is changed to static.